### PR TITLE
DEX-281 implement detection callback from Sage

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Annotation creation from detection response.
- As yet we cannot test this properly but by removing some checks (now restored) this has been tested to the point where annotations are created and insterted into the DB.

---

**REST API Updates**

A new example POST API is defined with its expected parameters, if needed to be specified for a thorough review

```
[POST] /api/v1/asset_group/sighting/{sighting_guid}/sage_detected/{job_uuid}
{
    See : https://kaiju.dyn.wildme.io:5005/api/engine/job/result/?jobid=f3155b31-3170-45de-a6d9-874716cc7e65&__format__=true
}
```

### CODE REVIEW

**Review by colinwkingen — COMMENTED** [review]
One question but looks good otherwise. I don't expect us to _always_ have a specified orientation. @naknomum Does that track with you are planning?

File: app/modules/annotations/models.py [hunk 1]
```diff
@@ -132,3 +133,20 @@ def validate_bounds(cls, bounds):
...
+        ytl = input_data.get('ytl')
+        width = input_data.get('width')
+        height = input_data.get('height')
+        theta = input_data.get('theta')
+
+        if xtl is None or ytl is None or width is None or height is None or theta is None:
+            raise HoustonException(
+                log_message=f'{input_data} missing fields',
+                message='input Data needs xtl, ytl, width, height and theta',
+            )
```
[the remark this answers is not in the document]
Comment (naknomum) [reply app/modules/annotations/models.py:149]: theta is "optional" now.  which means we assume it is 0 if not provided.   we could force that set to 0 if not provided or just leave it gone.

[the remark this answers is not in the document]
Comment (hwindsor) [reply app/modules/annotations/models.py:149]: I don't like either-or options, I find 
if whim:
     do one thing
else:
    do another 
usually ends badly :)
Implemented as setting it as 0 for now until you guys decide what you actually want this to do :)

**Comment by hwindsor** [comment]
Code coverage is low as it's not actually possible to properly test this until Sage implements the detection. I've tested it by removing some of the tests on validity of the input so it ok to merge. 
You can trust me (used car salesman smile)

File: app/modules/annotations/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (colinwkingen) [reply app/modules/annotations/models.py]: So your preference would be to require theta for annotation creation and to provide 0 as an input value in scenarios where there was no theta available?

[the remark this answers is not in the document]
Comment (hwindsor) [reply app/modules/annotations/models.py]: Jon gave a couple of options. I chose one of them, or was that question to @naknomum ?

[the remark this answers is not in the document]
Comment (colinwkingen) [reply app/modules/annotations/models.py]: Ah, reading comprehension issue on my side. I thought you didn't like either option but was incorrect. Carry on.

**Review by colinwkingen — APPROVED** [review]
LGTM.

**Pull request merged**